src/views/MarketFilter.tsx

```diff
@@ -37,50 +40,85 @@ export const MarketFilter = ({
   const stringGetter = useStringGetter();
   const navigate = useNavigate();
   const { hasPotentialMarketsData } = usePotentialMarkets();
+  const { uiRefresh, pml: showLaunchMarkets } = testFlags;
   const showProposeButton = hasPotentialMarketsData && !hideNewMarketButton;
 
+  const filterToggles = (
+    <$ToggleGroup
+      items={
+        Object.values(filters).map((value) => ({
+          label: stringGetter({ key: MARKET_FILTER_OPTIONS[value].label, fallback: value }),
+          slotAfter: MARKET_FILTER_OPTIONS[value]?.isNew && (
+            <NewTag>{stringGetter({ key: STRING_KEYS.NEW })}</NewTag>
+          ),
+          value,
+        })) as MenuItem<MarketFilters>[]
+      }
+      value={selectedFilter}
+      onValueChange={onChangeFilter}
+      overflow={uiRefresh ? 'wrap' : 'scroll'}
+    />
+  );
+
+  const launchMarketButton = uiRefresh ? (
+    <Button
+      onClick={() => navigate(`${AppRoute.Markets}/${MarketsRoute.New}`)}
+      size={ButtonSize.XSmall}
+      shape={ButtonShape.Pill}
+      action={ButtonAction.Primary}
+    >
+      {stringGetter({ key: STRING_KEYS.LAUNCH_MARKET_WITH_PLUS })}
+    </Button>
+  ) : (
+    <Button
+      onClick={() => navigate(`${AppRoute.Markets}/${MarketsRoute.New}`)}
+      size={ButtonSize.Small}
+    >
+      {stringGetter({ key: STRING_KEYS.PROPOSE_NEW_MARKET })}
+    </Button>
+  );
+
   return (
-    <$MarketFilter $compactLayout={compactLayout}>
-      <$SearchInput
-        placeholder={stringGetter({ key: searchPlaceholderKey })}
-        onTextChange={onSearchTextChange}
-      />
-
-      <div tw="row overflow-x-scroll">
-        <$ToggleGroupContainer $compactLayout={compactLayout}>
-          <$ToggleGroup
-            items={Object.values(filters).map((value) => ({
-              label: stringGetter({ key: MARKET_FILTER_OPTIONS[value].label, fallback: value }),
-              slotAfter: MARKET_FILTER_OPTIONS[value]?.isNew && (
-                <NewTag>{stringGetter({ key: STRING_KEYS.NEW })}</NewTag>
-              ),
-              value,
-            }))}
-            value={selectedFilter}
-            onValueChange={onChangeFilter}
-          />
-        </$ToggleGroupContainer>
-
-        {showProposeButton && (
-          <Button
-            onClick={() => navigate(`${AppRoute.Markets}/${MarketsRoute.New}`)}
-            size={ButtonSize.Small}
-          >
-            {stringGetter({ key: STRING_KEYS.PROPOSE_NEW_MARKET })}
-          </Button>
-        )}
+    <$MarketFilter $compactLayout={compactLayout} uiRefreshEnabled={uiRefresh}>
+      <div tw="flex items-center gap-0.5">
+        <$SearchInput
+          placeholder={stringGetter({ key: searchPlaceholderKey })}
+          onTextChange={onSearchTextChange}
+        />
+        {uiRefresh && showProposeButton && showLaunchMarkets && launchMarketButton}
```

Comment:
is adding the `showLaunchMarkets` to determine whether to show propose button an intentional change? since we used to just show the button when `showProposeButton` is true

Reply:
yess - so was following logic for when we enable the Launch Market route [here](https://github.com/dydxprotocol/v4-web/blob/main/src/App.tsx#L151) - since the route doesn't exist unless pml is enabled, I also hide it here. Note that the existing "Propose markets" button logic doesn't change